src/linter/engine.mjs

```diff
@@ -3,9 +3,12 @@
 /**
  * Creates a linter engine instance to validate ApiDocMetadataEntry entries
  *
- * @param {import('./types').LintRule} rules Lint rules to validate the entries against
+ * @param {{
+ * multiEntryRules: import('./types').MultipleEntriesLintRules[]
```

Comment:
I think all rules could be a multi-entry rule, the singleEntryRules can just loop all entries.

Reply:
That makes a lot more sense than what I had setup, I'll adjust it to that, thanks!